### DISCUSSION TITLE
How to interact user directories and files

### DISCUSSION THREAD

**Discussion by 0x4a61636f62** [glacambre/firenvim#1561] [Q&A] [answered]
In my Neovim configuration, I use commands to retrieve the directory paths, file contents etc.

For example, here are some functions and keymappings I use:
```
-- Partial command input (':r !')
map('n', '<S-1>', [[<Cmd>lua vim.api.nvim_feedkeys(vim.api.nvim_replace_termcodes(':r !', true, false, true), 'n', false)<CR>]], { noremap = true, silent = true })

-- Function to insert the current file path
function insert_current_file_path()
  local current_file_path = vim.fn.fnamemodify(vim.fn.expand("%:p"), ":~")
  vim.api.nvim_put({current_file_path}, 'l', true, true)
end

-- Insert current file path into buffer
map('n', '<Leader>p', [[<Cmd>lua insert_current_file_path()<CR>]], { noremap = true, silent = true })

-- Function to insert the tree of the current file's directory
function insert_tree_of_current_file_dir()
  local current_file_dir = vim.fn.expand('%:h')
  local cmd = "tree " .. current_file_dir
  vim.cmd("r !" .. cmd)
end

map('n', '<leader>t', [[<Cmd>lua insert_tree_of_current_file_dir()<CR>]], { noremap = true, silent = true })
```

Having the same capability in **Firenvim**would be very useful, but I have yet to find a way to do it. I want to change the directory with `:cd ~` and then read the output of some command with `':r !'`.

It seems we are limited to a working file, e.g., `/run/user/1000/firenvim/some.domain.com__TEXTAREA-id-prompt-textarea_2023-09-01T17-52-46-067Z.txt`. 

@glacambre and the community, do you have any ideas how it can be done?

**Comment by glacambre** [comment, marked as answer]
@JacobStraberg I very regularly do `:cd ~/some/dir` followed by a `:r!some --command` and it works fine (that's how I insert Firenvim's changelogs on the mozilla add-on publishing page for example). Could you provide the commands you tried, their expected outputs and what you got instead?

**Reply by 0x4a61636f62** [reply]
@glacambre I am not sure what I did wrong, but indeed it works to do `:cd ~/some/dir` and reading outputs of commands.
